addons/website_forum/models/forum.py

```diff
@@ -128,11 +131,28 @@ def _get_default_faq(self):
     karma_post = fields.Integer(string='Ask questions without validation', default=100)
     karma_moderate = fields.Integer(string='Moderate posts', default=1000)
 
+    @api.depends('post_ids')
+    def _compute_last_post_id(self):
+        for forum in self:
+            forum.last_post_id = forum.post_ids.search([('parent_id', '=', False)], order='create_date desc', limit=1)
```

Comment:
in general, we consider that if you have 20 forums,
you will do 20 queries while with a read group, you will have only one request.

But in this case, seems like 100 queries are faster than 1 group by, so we can just keep it.

The problem is that in case of 1 big forum (the main) and a lot of small (elearning), the group by forum_id will do extra extra cost and idx ignored.

https://explain.depesz.com/s/HhU4 (select max(id) without group by)
https://explain.depesz.com/s/SPSX (select max(id) with group by)

Reply:
@Esperax still, I doubt you tested this... `forum.post_ids.search` ca ne va pas search sur les `post_ids` du forum mais sur tous les posts de tous les forums... c'est égal à `self.env['forum.post'].search`. C'est ok de ne pas tout savoir mais visiblement, tu n'as pas testé sinon tu verrais direct que ca ne marche pas... ou est-ce que je me trompe?

Reply:
Yeah, I knew about that, but I didn't fully tested the thing, so I didn't realize.
Should I create python tests for this field ? To check, with 2 forum that they have a different last_post_id ?

Reply:
Seems like you would want to test that if you put the value 'x' in a field, there is a 'x' value inside to me... so no I don't see why that would be necessary. JKE correct me if I am wrong.

Reply:
as qsm say, nothing to test except yourself your code.
But if you want to write some tests,  you can  test the performance of forum page, before and after your code ;)